src/transformation/visitors/binary-expression/assignments.ts

```diff
@@ -1,15 +1,44 @@
 import * as ts from "typescript";
 import * as lua from "../../../LuaAST";
-import { cast, castEach } from "../../../utils";
+import { cast } from "../../../utils";
 import { TransformationContext } from "../../context";
 import { isTupleReturnCall } from "../../utils/annotations";
 import { validateAssignment, validatePropertyAssignment } from "../../utils/assignment-validation";
+import { createExportedIdentifier, getDependenciesOfSymbol, isSymbolExported } from "../../utils/export";
 import { createImmediatelyInvokedFunctionExpression, createUnpackCall, wrapInTable } from "../../utils/lua-ast";
 import { LuaLibFeature, transformLuaLibFunction } from "../../utils/lualib";
 import { isArrayType, isDestructuringAssignment } from "../../utils/typescript";
 import { transformElementAccessArgument } from "../access";
 import { isArrayLength, transformDestructuringAssignment } from "./destructuring-assignments";
 
+export function transformAssignmentLeftHandSideExpression(
+    context: TransformationContext,
+    node: ts.Expression
+): lua.AssignmentLeftHandSideExpression {
+    const symbol = context.checker.getSymbolAtLocation(node);
+    const left = context.transformExpression(node);
+
+    return lua.isIdentifier(left) && symbol && isSymbolExported(context, symbol)
+        ? createExportedIdentifier(context, left)
+        : cast(left, lua.isAssignmentLeftHandSideExpression);
+}
+
+export function createNestedImmediatelyInvokedFunctionExpressionsForEachSymbolSideEffect(
+    context: TransformationContext,
+    rootAssignment: lua.AssignmentStatement,
+    symbols: ts.Symbol[]
+): lua.AssignmentStatement {
+    return symbols.reduce<lua.AssignmentStatement>((statement, symbol) => {
+        const [left] = statement.left;
+        const assignmentFunction = createImmediatelyInvokedFunctionExpression([statement], left);
+        const identifierToAssign = createExportedIdentifier(
+            context,
+            lua.createIdentifier(symbol.escapedName.toString())
```

Comment:
Escaped string shouldn't be used as an unescaped one, there should be an unescaped one under `name` or `text` name